dashboard/models/utils.py

```diff
@@ -81,3 +83,108 @@ def schedule_email(email_func, input_args, input_kwargs=None):
     scheduler.add_job(uuid4().hex, email_func, trigger='date',
                       run_date=datetime.now(), args=input_args,
                       kwargs=input_kwargs)
+
+
+def update_xnat_usability(scan, app_config):
+    """Update XNAT usability data for a scan.
+
+    Args:
+        scan (:obj:`dashboard.models.Scan`): The series to push QC data for.
+        app_config (:obj:`dict`): Configuration of the current app instance,
+            as retrieved from current_app.config
+    """
+    study = scan.get_study()
+    site_settings = study.sites[scan.session.site.name]
+
+    exp_name = getattr(
+        scan.session, site_settings.xnat_convention.lower() + "_name"
+    )
+
+    user, password = get_xnat_credentials(site_settings, app_config)
+
+    if scan.flagged():
+        quality = 'questionable'
+    elif scan.blacklisted():
+        quality = 'unusable'
+    else:
+        quality = 'usable'
+
+    async_xnat_update(
+        site_settings.xnat_url,
+        user,
+        password,
+        site_settings.xnat_archive,
+        exp_name,
+        scan.series,
+        scan.qc_review.comment,
+        quality
+    )
+
+
+def get_xnat_credentials(site_settings, app_config):
+    """Retrieve the xnat username and password for a given study/site.
+
+    Args:
+        site_settings (:obj:`dashboard.models.StudySite`): A StudySite record
+        app_config (:obj:`dict`): Configuration of the current app instance,
+            as retrieved from current_app.config
+    """
+    if app_config.get('XNAT_USER'):
```

Comment:
Should this be `XNAT_ENABLED`?

Reply:
Nah. This function should only be run if its already been checked that xnat is enabled, but the function should check for (and return) a username/pass combo found in the env before looking for a file